src/main/java/org/tikv/raw/RawKVClient.java

```diff
@@ -152,20 +153,48 @@ public ByteString putIfAbsent(ByteString key, ByteString value) {
    * @param key key
    * @param value value
    * @param ttl TTL of key (in seconds), 0 means the key will never be outdated.
-   * @return a ByteString. returns ByteString.EMPTY if the value is written successfully. returns
-   *     the previous key if the value already exists, and does not write to TiKV.
+   * @return a ByteString. returns Optional.EMPTY if the value is written successfully. returns the
+   *     previous key if the value already exists, and does not write to TiKV.
    */
-  public ByteString putIfAbsent(ByteString key, ByteString value, long ttl) {
-    String label = "client_raw_put_if_absent";
+  public Optional<ByteString> putIfAbsent(ByteString key, ByteString value, long ttl) {
```

Comment:
@birdstorm can the customer accept this API change?

Reply:
That is not very easy I think. We need to document all the changes so that they can update their code.